pkg/api/validate/dynamic/sku.go

```diff
@@ -0,0 +1,71 @@
+package dynamic
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/computeskus"
+)
+
+// ValidateWorkerSku uses resourceSkusClient to ensure that the VM sizes listed in the cluster document are available for use in the target region.
+func (dv *dynamic) ValidateVmSku(ctx context.Context, location string, subscriptionID string, oc *api.OpenShiftCluster) error {
+	// Get a list of available worker SKUs, filtering by location.
+	filter := fmt.Sprintf("location eq %s", location)
+	skus, err := dv.resourceSkusClient.List(ctx, filter)
```

Comment:
Forgot to mention this: I think we need a quite verbose comment explaining why we can not use `dv.env.VMSku(string(VMSize))` here which is already present.

Someone might want to try and optimise this by replacing this call with `dv.env.VMSku(string(VMSize))` it we will be performing wrong validation.